src/main/kotlin/org/rust/ide/typing/RsAngleBraceHandlers.kt

```diff
@@ -0,0 +1,136 @@
+package org.rust.ide.typing
+
+import com.intellij.codeInsight.CodeInsightSettings
+import com.intellij.codeInsight.editorActions.TypedHandlerDelegate
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.ex.EditorEx
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import com.intellij.openapi.fileTypes.FileType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.tokenSetOf
+
+private val GENERIC_NAMED_ENTITY_KEYWORDS = tokenSetOf(RsElementTypes.FN, RsElementTypes.STRUCT,
+    RsElementTypes.ENUM, RsElementTypes.TRAIT, RsElementTypes.TYPE_KW)
```

Comment:
For `ElementType`s, we usually prefer unqualified form: `tokenSetOf(FN, STRUCT, ...)`.